packages/solana/src/client.ts

```diff
@@ -0,0 +1,482 @@
+      this.resetWcConnection()
+      this.resetNetwork()
+    }
+  }
+
+  private syncProfile(_address: string) {
+    this.setProfileName(null)
+    this.setProfileImage(null)
+  }
+
+  private async syncNetwork(chainImages?: Web3ModalClientOptions['chainImages']) {
+    const address = SolStoreUtil.state.address
+    const storeChainId = SolStoreUtil.state.caipChainId
+    const isConnected = SolStoreUtil.state.isConnected
+
+    if (this.chains) {
+      const chain = SolHelpersUtil.getChainFromCaip(this.chains, storeChainId)
+      if (chain) {
+        const caipChainId: CaipNetworkId = `${chain.name}:${chain.chainId}`
+
+        this.setCaipNetwork({
+          id: caipChainId,
+          name: chain.name,
+          imageId: PresetsUtil.EIP155NetworkImageIds[chain.chainId],
+          imageUrl: chainImages?.[chain.chainId]
+        })
+        if (isConnected && address) {
+          if (chain.explorerUrl) {
+            const url = `${chain.explorerUrl}/account/${address}`
+            this.setAddressExplorerUrl(url)
+          } else {
+            this.setAddressExplorerUrl(undefined)
+          }
+          if (this.hasSyncedConnectedAccount) {
+            this.syncProfile(address)
+            await this.syncBalance(address)
+          }
+        }
+      }
+    }
+  }
+
+  private async syncBalance(address: string) {
+    const caipChainId = SolStoreUtil.state.caipChainId
+    if (caipChainId && this.chains) {
+      const chain = SolHelpersUtil.getChainFromCaip(this.chains, caipChainId)
+      if (chain) {
+        const balance = await this.WalletConnectConnector.getBalance(address)
+        this.setBalance(balance.decimals.toString(), chain.currency)
+      }
+    }
+  }
+
+  private syncRequestedNetworks(
+    chains: Web3ModalClientOptions['chains'],
+    chainImages?: Web3ModalClientOptions['chainImages']
+  ) {
+    const requestedCaipNetworks = chains?.map(
+      chain =>
+        ({
+          id: `${chain.name}:${chain.chainId}`,
+          name: chain.name,
+          imageId: PresetsUtil.EIP155NetworkImageIds[chain.chainId],
+          imageUrl: chainImages?.[chain.chainId]
+        }) as CaipNetwork
+    )
+    this.setRequestedCaipNetworks(requestedCaipNetworks ?? [])
+  }
+
+  public async switchNetwork(caipNetwork: CaipNetwork) {
+    const caipChainId = caipNetwork.id
+
+    const providerType = SolStoreUtil.state.providerType
+
+    const chain = SolHelpersUtil.getChainFromCaip(this.chains, caipChainId ?? '')
+
+    if (this.chains) {
+      if (chain) {
+        SolStoreUtil.setCaipChainId(`${chain.name}:${chain.chainId}`)
+        SolStoreUtil.setCurrentChain(chain)
+        localStorage.setItem(SolConstantsUtil.CAIP_CHAIN_ID, `${chain.name}:${chain.chainId}`)
+        if (providerType?.includes(ConstantsUtil.INJECTED_CONNECTOR_ID)) {
+          const wallet =
+            providerType.split('_')[1] === 'Trust'
+              ? 'trustWallet'
+              : (providerType.split('_')[1] as AdapterKey)
```

Comment:
same as before, if not possible to manipulate the name, would add some util function to prevent duplicated code.